python_liftbridge/python_liftbridge.py

```diff
@@ -64,8 +74,19 @@ def _fetch_metadata(self, metadata_request):
         return response
 
     @handle_rpc_errors_in_generator
-    def _subscribe(self, subscribe_request):
-        for message in self.stub.Subscribe(subscribe_request):
+    def _subscribe(self, subscribe_request, timeout=None):
+        # The first message in a subscription tells us if the subscribe succeeded.
+        # From the docs:
+        # """When the subscription stream is created, the server sends an
+        # empty message to indicate the subscription was successfully created.
+        # Otherwise, an error is sent on the stream if the subscribe failed.
+        # This handshake message must be handled and should not be exposed
+        # to the user."""
+        subscription = self.stub.Subscribe(subscribe_request, timeout)
```

Comment:
Just to make sure, does `Subscribe()` accept timeout as `None` ?

Reply:
I refer [here](https://github.com/dgzlopes/python-liftbridge/blob/1f1e74c57a5edcde0cd140c530675ad8fc458005/python_liftbridge/base.py#L36)
```python

    def _connect(self):
        try:
            grpc.channel_ready_future(
                self.channel,
            ).result(timeout=self.timeout)
        except grpc.FutureTimeoutError:
            sys.exit('Error connecting to server')
        else:
            return python_liftbridge.api_pb2_grpc.APIStub(self.channel)
```

it seems that `self.stub` returns `APIStub` object.
Refer to [`APISub`] (https://github.com/dgzlopes/python-liftbridge/blob/1f1e74c57a5edcde0cd140c530675ad8fc458005/python_liftbridge/api_pb2_grpc.py#L7)

it seems not

```python

# in APIStub

        self.Subscribe = channel.unary_stream(
                '/proto.API/Subscribe',
                request_serializer=protos_dot_api__pb2.SubscribeRequest.SerializeToString,
                response_deserializer=protos_dot_api__pb2.Message.FromString,
                )
```

Did I miss something ?

Reply:
There is the `Lift` timeout and the gRPC timeout which currently is only being added for `subscribe`. The "client" or `Lift` timeout code doesn't change with this PR.

The `subscribe` defaults to `timeout=None` which means it just kind of blocks and hangs around on the next message. That was the old default behaviour as well.